en/dev_log/ulog_file_format.md

```diff
@@ -41,233 +48,331 @@ The file consists of three sections:
+  - The `key` part has the format identical as the field in the [Format message](#f-format-message) (can also be a custom type), but consists of only a single field without ending `;`, eg. `char[5] sys_toolchain_ver`
+  - The `value` contains the data corresponding to the `key`, and is written right after the `key`, e.g. `9.4.0`.
+
+:::note
+A key : value pair defined in the Information message should be unique. Meaning there shouldn't be more than one definition with the same key value!
+:::
+
+Parsers can store information messages as a dictionary.
+
+Predefined information messages are:
 
 key | Description | Example for value
 --- | ---  | ---
-char[value_len] sys_name | Name of the system |  "PX4"
-char[value_len] ver_hw | Hardware version (board) |  "PX4FMU_V4"
-char[value_len] ver_hw_subtype| Board subversion (variation)|  "V2"
-char[value_len] ver_sw | Software version (git tag)|  "7f65e01"
-char[value_len] ver_sw_branch| git branch |  "master"
-uint32_t ver_sw_release | Software version (see below)|  0x010401ff
-char[value_len] sys_os_name | Operating System Name |  "Linux"
-char[value_len] sys_os_ver | OS version (git tag) |  "9f82919"
-uint32_t ver_os_release | OS version (see below) |  0x010401ff
-char[value_len] sys_toolchain | Toolchain Name |  "GNU GCC"
-char[value_len] sys_toolchain_ver | Toolchain Version |  "6.2.1"
-char[value_len] sys_mcu | Chip name and revision |  "STM32F42x, rev A"
-char[value_len] sys_uuid | Unique identifier for vehicle (eg. MCU ID) |  "392a93e32fa3"...
-char[value_len] log_type | Type of the log (full log if not specified) | "mission"
-char[value_len] replay | File name of replayed log if in replay mode | "log001.ulg"
-int32_t time_ref_utc | UTC Time offset in seconds | -3600 |
-
-  The format of `ver_sw_release` and `ver_os_release` is: 0xAABBCCTT, where AA is major, BB is minor, CC is patch and TT is the type.
-  Type is defined as following: `>= 0`: development, `>= 64`: alpha version, `>= 128`: beta version, `>= 192`: RC version, `== 255`: release version.
-  So for example 0x010402ff translates into the release version v1.4.2.
-
-  This message can also be used in the Data section (this is however the preferred section).
-
-
-- 'M': information message multi.
-  ```c
-  struct ulog_message_info_multiple_header_s {
-  	struct message_header_s header;
-  	uint8_t is_continued; ///< can be used for arrays
-  	uint8_t key_len;
-  	char key[key_len];
-  	char value[header.msg_size-2-key_len]
-  };
-  ```
-  The same as the information message, except that there can be multiple messages with the same key (parsers store them as a list).
-  The `is_continued` can be used for split-up messages: if set to 1, it is part of the previous message with the same key.
-  Parsers can store all information multi messages as a 2D list, using the same order as the messages occur in the log.
+`char[3] sys_name` | Name of the system |  "PX4"
```

Comment:
I find this more confusing than before. The key is generic, but now it's tied to the example

Reply:
I found the `value_len` part confusing when I first encountered the doc. Is there a better way to phrase it?

Since currently with the example, it is very clear that this is how you can use the key : value pair, but it's also not generic anymore. Maybe add more text above to state that it can be generic?

Reply:
@bkueng I think you are saying that the values @junwoo091400 has suggested are length values specific to PX4 implementation, and that some other flight stack using this spec might use different numbers. Is that right?

There are two possible fixes. 
1. We use `value_len` but we have a note at the bottom stating (something like):
   > `value_len` represents the data size required for storing the key. This will be key and system specific.
2. Use @junwoo091400 values, but at the end have a note:
  > The values for the value data-size in the table above are those used by PX4. Other systems may use other values.
  
I tend to think option 1 is better for a "generic spec" - because with option 2 people are more likely just to see the values and assume they can be hard coded (i.e. not read the spec properly). So I'd revert and use option "1" (assuming Beat agrees)